### DISCUSSION TITLE
❓ Questions / Help / Support  Unexpected WER values on CommonVoice Test

### DISCUSSION THREAD

**Discussion by kfmn** [snakers4/silero-models#152] [Q&A] [answered]
Hi, I am trying Silero English STT model to recognize a random subset (100 files) from English Common Voice Test. 
I use a code based on a snippet from your README file, namely
```

device = torch.device('cpu')  # gpu also works, but our models are fast enough for CPU

def transcribe_file(filename, model, decoder, utils):
    (read_batch, split_into_batches, read_audio, prepare_model_input) = utils  # see function signature for details
    test_files = glob(filename)
    batches = split_into_batches(test_files, batch_size=10)
    input = prepare_model_input(read_batch(batches[0]),
                            device=device)

    output = model(input)
    result=''
    for example in output:
        result += decoder(example.cpu()) + ' '
    return result


def transcribe_dir(wavdir, recfile, model, decoder, utils):
    with codecs.open(recfile, 'w', encoding='utf8') as f:
        for speech_file in os.listdir(wavdir):
            print(speech_file)
            result = transcribe_file(os.path.join(wavdir,speech_file), model, decoder, utils)
            fname=speech_file.split('/')[-1]
            uttid=fname[:-4]      # cut .wav
            if not (result=='<error>' or result=='<noresult>'):
                f.write(f'{uttid} {result}\n')



dirin = sys.argv[1]
lang = sys.argv[2]        # available values: 'en', 'es', 'dе'
result = sys.argv[3]


model, decoder, utils = torch.hub.load(repo_or_dir='snakers4/silero-models',
                                       model='silero_stt',
                                       language=lang,
                                       device=device)


transcribe_dir(dirin, result, model, decoder, utils)
```
I obtain WER of about 35% which differs substantially from that reported in you diagram within README file (less than 20%). This difference cannot be explained by selection bias or large confidence interval...
Please explain, 
- what English model is used by default ?
- where can I found metrics for it ?
- how can I use more accurate model to obtain results presented in your diagram ?

**Comment by snakers4** [comment, marked as answer]
Which model are you using?
If you are using the community edition small models, looks roughly [similar](https://github.com/snakers4/silero-models/wiki/Quality-Benchmarks#en-v6).

**Reply by kfmn** [reply]
I use a model that was downloaded by default while calling

```
model, decoder, utils = torch.hub.load(repo_or_dir='snakers4/silero-models',
                                       model='silero_stt',
                                       language=lang,
                                       device=device)

```

How can I load and use more advanced models ?

**Reply by snakers4** [reply]
Larger models can be loaded via `jit_model` param:

https://github.com/snakers4/silero-models/blob/f188d86306aace31526a050dd3533c6d91094b3d/src/silero/silero.py#L5-L40

Available models:

https://github.com/snakers4/silero-models/blob/f188d86306aace31526a050dd3533c6d91094b3d/models.yml#L4-L23

**Reply by kfmn** [reply]
Thanks a lot, will try this soon. 
I have one more question if possible... I would like to test Russian STT as well. But there are no Russian models here. I have found that there is some Russian STT API at https://www.silero.ai/ but I don't understand what should I do to obtain API token which is required field for the payload description. Could you please help me with this?

**Reply by snakers4** [reply]
Russian models are fully private and available only for commercial customers.

**Reply by kfmn** [reply]
OK, can I just test them somehow (I can pay some fee) to decide if I need to buy them for my tasks?
Maybe you have some trial access or so...

**Reply by snakers4** [reply]
Please contact us via email or via telegram

**Reply by kfmn** [reply]
Ok, thank you

**Reply by kfmn** [reply]
Sorry to trouble you again. I tried to replace

```
model, decoder, utils = torch.hub.load(repo_or_dir='snakers4/silero-models',
                                       model='silero_stt',
                                       language=lang,.
                                       device=device)
```

with

```
model, decoder, utils = silero_stt(language=lang,.
                version='latest',.
                jit_model='jit')
```

as you advised. Model has been downloaded but then an exception raised:

Traceback (most recent call last):
  File "test_pytorch_jit.py", line 46, in <module>
    model, decoder, utils = silero_stt(language=lang,
  ....
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory

How can I fix this?

**Reply by snakers4** [reply]
The "Minimal example" from colab with large model works:

![image](https://user-images.githubusercontent.com/12515440/165937666-5574f159-1319-4896-ba0c-bc067b0dfa4a.png)

**Reply by kfmn** [reply]
Upd: I have found the downloaded model, but when trying to look into I get the following message:

End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.

It seems like model archive is broken...

**Reply by kfmn** [reply]
You are right, 'jit_xlarge' works properly, but simpy 'jit' causes trouble I described. Please check it.
And WER for jit_xlarge is about 27% which is much better than 35 but still much worse than you reported. Does it mean I am still using Community Edition model, not Enterprise Edition ? Is it possible to test EE too ?

**Reply by snakers4** [reply]
Yes
We reserve EE test for commercial clients

**Reply by kfmn** [reply]
Ok, got it
